core/src/main/codegen/templates/Parser.jj

```diff
@@ -937,6 +942,45 @@ SqlNode SqlQueryEof() :
     { return query; }
 }
 
+/**
+ * Parses a list of SQL statements separated by semicolon.
+ * The semicolon is required between statements, but is
+ * optional at the end.
+ */
+SqlNodeList SqlStmtList() :
+{
+    final List<SqlNode> stmtList = new ArrayList<SqlNode>();
+    SqlNode stmt;
+}
+{
+    stmt = SqlStmtWithSemiColon() {
+        stmtList.add(stmt);
+    }
```

Comment:
In this statement, the follow sql  may pass the parser:
`select * from a   select * from b`
There is no ';' between the sql statements.

Reply:
I run the test and I find that It will throw exception no matter with or without this PR as follows:
```
org.apache.calcite.sql.parser.SqlParseException: Encountered "SELECT" at line 1, column 19.
Was expecting one of:
    <EOF> 
    "ORDER" ...
    ...

  at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
  at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
  at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
  at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
  at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: org.apache.calcite.sql.parser.impl.ParseException: Encountered "SELECT" at line 1, column 19.
Was expecting one of:
    <EOF> 
    ...
    
  at org.apache.calcite.sql.parser.impl.SqlParserImpl.generateParseException(SqlParserImpl.java:23129)
  at org.apache.calcite.sql.parser.impl.SqlParserImpl.jj_consume_token(SqlParserImpl.java:22946)
  at org.apache.calcite.sql.parser.impl.SqlParserImpl.SqlStmtEof(SqlParserImpl.java:914)
  at org.apache.calcite.sql.parser.impl.SqlParserImpl.parseSqlStmtEof(SqlParserImpl.java:184)
  at org.apache.calcite.sql.parser.SqlParser.parseQuery(SqlParser.java:148)
  ... 28 more
```

Reply:
Yes, the `parseSqlStmtList() ` must end with  a `<EOF>` token.